Simplenote/Classes/NoteInfoViewController.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  Created by Kevin LaCoste on 2020-05-15.
+//  Copyright © 2020 Automattic. All rights reserved.
+//
+
+import UIKit
+
+class NoteInfoViewController: UIViewController, UITableViewDataSource {
+    
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var dismissButton: UIButton!
+    @IBOutlet private weak var tablewView: UITableView!
+    
+    @objc var dismissAction: (() -> Void)?
+    @objc var note: Note!
```

Comment:
`NoteInfoViewController` has a strong dependency towards `Note` (that is, the VC shouldn't even exist if there's no note). 

Although this approach works as well (the app will crash if the `note` is not set), you can actually improve this by requiring the Note via initializer:

```
init(note: Note) {
    self.note = note
    super.init(....
}
```

### Why that'd be an improvement:
Imagine there's not one, but multiple strong dependencies. In order to figure out which public properties **must** be set, you could:

1. Read thru the code. **Problem:** Takes too long!
2. Read the class documentation. **Problem:** Nobody really documents that (including this class!). Plus where should it go, in the property itself? at the top of the class?
3. Read the designated initializer

(3) is a super natural way to spot the strong requirements. The SDK itself is pretty much full of such examples. For instance, [NSPersistentStoreCoordinator](https://developer.apple.com/documentation/coredata/nspersistentstorecoordinator) has a single public initializer, which requires a **NSManagedObjectModel**.

### Why it's different from the IBOutlets approach:
IBOutlets themselves are set by the `xib` / `storyboard` unarchiver. Even if we wanted to, there's no way to easily hint there's something missing.